src/Aspire.Hosting/SqlServer/SqlServerBuilderExtensions.cs

```diff
@@ -26,37 +26,65 @@ public static IResourceBuilder<SqlServerContainerResource> AddSqlServerContainer
         var sqlServer = new SqlServerContainerResource(name, password);
 
         return builder.AddResource(sqlServer)
-                      .WithManifestPublishingCallback(WriteSqlServerContainerToManifest)
+                      .WithManifestPublishingCallback(context => WriteSqlServerContainerToManifest(context, sqlServer))
                       .WithAnnotation(new ServiceBindingAnnotation(ProtocolType.Tcp, port: port, containerPort: 1433))
                       .WithAnnotation(new ContainerImageAnnotation { Registry = "mcr.microsoft.com", Image = "mssql/server", Tag = "2022-latest" })
                       .WithEnvironment("ACCEPT_EULA", "Y")
-                      .WithEnvironment("MSSQL_SA_PASSWORD", sqlServer.Password);
+                      .WithEnvironment(context =>
+                      {
+                          if (context.PublisherName == "manifest")
+                          {
+                              context.EnvironmentVariables.Add("MSSQL_SA_PASSWORD", $"{{{sqlServer.Name}.inputs.password}}");
+                          }
+                          else
+                          {
+                              context.EnvironmentVariables.Add("MSSQL_SA_PASSWORD", sqlServer.Password);
+                          }
+                      });
+
     }
 
     /// <summary>
-    /// Adds a SQL Server connection to the application model. Connection strings can also be read from the connection string section of the configuration using the name of the resource.
+    /// Adds a SQL Server resource to the application model. A container is used for local development.
     /// </summary>
     /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
     /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
-    /// <param name="connectionString">The connection string.</param>
-    /// <returns>A reference to the <see cref="IResourceBuilder{SqlServerConnectionResource}"/>.</returns>
-    public static IResourceBuilder<SqlServerConnectionResource> AddSqlServerConnection(this IDistributedApplicationBuilder builder, string name, string? connectionString = null)
+    /// <returns>A reference to the <see cref="IResourceBuilder{SqlServerContainerResource}"/>.</returns>
+    public static IResourceBuilder<SqlServerServerResource> AddSqlServer(this IDistributedApplicationBuilder builder, string name)
     {
-        var sqlServerConnection = new SqlServerConnectionResource(name, connectionString);
+        var password = Guid.NewGuid().ToString("N") + Guid.NewGuid().ToString("N").ToUpper();
```

Comment:
What is the reason to repeat the guid?

Reply:
The reason to repeat the GUID is that the default password Policy for the SQL container needs upper case and lower case characters with special cases. This is just a cheats way of using GUIDs to do that.